### PR TITLE
Fix exclusive hook conflict

### LINKED ISSUE #1038: FixDepletedItems and FixUnlimitedPotionsBug Tweaks conflict sharing AIActionItemCastSpell hook
Either have to rework one or the other, or combine them into a multifunction tweak of sorts.

### PR DESCRIPTION
**Pull request by GoLoT**
Fixes #1038 

I can't test it because I haven't updated my testing server to .16 yet. Can someone quickly test it?

### CODE REVIEW

File: Plugins/Tweaks/Tweaks/PreserveDepletedItems.cpp [hunk 1]
```diff
@@ -14,28 +14,31 @@ namespace Tweaks {
...
 PreserveDepletedItems::PreserveDepletedItems(Services::HooksProxy* hooker)
 {
-    pAIActionItemCastSpell_hook = hooker->RequestExclusiveHook
-        <Functions::_ZN12CNWSCreature21AIActionItemCastSpellEP20CNWSObjectActionNode>(&CNWSCreature__AIActionItemCastSpell_hook);
+    hooker->RequestSharedHook<Functions::_ZN12CNWSCreature21AIActionItemCastSpellEP20CNWSObjectActionNode, uint32_t>(&CNWSCreature__AIActionItemCastSpell_hook);
 }
 
 
-uint32_t PreserveDepletedItems::CNWSCreature__AIActionItemCastSpell_hook(CNWSCreature *pThis, CNWSObjectActionNode *pNode)
+void PreserveDepletedItems::CNWSCreature__AIActionItemCastSpell_hook(bool before, CNWSCreature *pThis, CNWSObjectActionNode *pNode)
```
Comment (plenarius) [inline Plugins/Tweaks/Tweaks/PreserveDepletedItems.cpp:23]: pThis unused parameter

Comment (GoLoT) [reply Plugins/Tweaks/Tweaks/PreserveDepletedItems.cpp]: Thanks for catching it.

**Comment by mtijanic** [comment]
Is this ready to merge (i.e. someone did the most basic test at least)?

**Comment by GoLoT** [comment]
I didn't. I'll update my testing environment eventually and create a simple test scenario but for now that will have to wait.

Since no one seems to be in a rush for this, I'd keep it open for now until either me or someone that needs the fix tests it.

**Comment by plenarius** [comment]
Finally got around to testing this as we don't use the PreserveDepletedItems tweak. All seems good, merging.

**Pull request merged**